Source/Core/Core/PowerPC/PowerPC.cpp

```diff
@@ -132,9 +132,16 @@ void DoState(PointerWrap& p)
   p.Do(ppcState.reserve_address);
 
   ppcState.iCache.DoState(p);
+  ppcState.dCache.DoState(p);
 
   if (p.IsReadMode())
   {
+    if (!ppcState.m_enable_dcache)
+    {
+      INFO_LOG_FMT(POWERPC, "Flushing data cache");
+      ppcState.dCache.FlushAll();
+    }
```

Comment:
The goal here is that if dcache was enabled when the state was saved, but disabled when the state was loaded, data from dcache will get moved into memory, right?

Is something needed in the opposite direction (disabled when the state was saved, but enabled when it was loaded)? I *think* the cache should always be all invalid in that case, but I'm not 100% sure of it.